recipe/test_nvcc.sh

```diff
@@ -64,8 +64,48 @@ if [[ "${CFLAGS}" == "${CFLAGS_CONDA_NVCC_TEST}" ]]; then
 else
     echo "CFLAGS is incorrectly set to '${CFLAGS}', should be set to '${CFLAGS_CONDA_NVCC_TEST}'" && exit 1
 fi
+
 # Reactivate
 source ${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh
 
 # Try building something
-nvcc test.cu
\ No newline at end of file
+nvcc test.cu
+
+# Try different CMake setups
+cd cmake-tests/
+
+set +ex
+exitcode=0
+count=0
+summary=""
+for flags in \
+    "-DWITH_ENABLE_LANGUAGE=OFF -DCUDA_FINDER=CUDA       " \
+    "-DWITH_ENABLE_LANGUAGE=OFF -DCUDA_FINDER=CUDAToolkit" \
+    "-DWITH_ENABLE_LANGUAGE=ON  -DCUDA_FINDER=CUDA       " \
+    "-DWITH_ENABLE_LANGUAGE=ON  -DCUDA_FINDER=CUDAToolkit" \
+    "-DWITH_ENABLE_LANGUAGE=ON  -DCUDA_FINDER=OFF        " \
+    ; do
+        ((count+=1))
+        echo -e "\n\n--------"
+        echo -e "Test #$count:" $flags
+        echo -e "--------\n\n"
+        rm -rf build || true
+        mkdir -p build
+        cd build
+        export CUDACXX=${CUDA_PATH}/bin/nvcc
```

Comment:
Hmm, I guess this is the catch-22? Without this it will find our wrapper and infer incorrect directories in the case of `enable_language(CUDA)` being used? I'll poke around and see if there's any way to get it to use our wrapper while still pointing in the correct place.

Alternatively, we can pass desired flags directly in `CUDACXX`: https://cmake.org/cmake/help/latest/envvar/CUDACXX.html

Reply:
Yep, I think so. I can push a commit without it so you can see what happens.

> Alternatively, we can pass desired flags directly in CUDACXX

This can work nicely on Linux, but quotes within variables on Windows is such a PITA and has biten us before.

Reply:
I don't think it will work for Linux either unfortunately. Seems like CMake parses it and rips out the `--ccbin=...` if you pass it so if someone invokes `CMAKE_CUDA_COMPILER` it won't have the option

Reply:
I have commented out the CUDACXX definitions on 59f1cf3 if you want take a look.